packages/cli/src/services/abstract-check-runner.ts

```diff
@@ -116,38 +118,47 @@ export default abstract class AbstractCheckRunner extends EventEmitter {
     socketClient.on('message', (topic: string, rawMessage: string|Buffer) => {
       const message = JSON.parse(rawMessage.toString('utf8'))
       const topicComponents = topic.split('/')
-      const checkRunId = topicComponents[4]
-      const subtopic = topicComponents[5]
+      const sequenceId = topicComponents[4]
+      const checkRunId = topicComponents[5]
+      const subtopic = topicComponents[6]
 
-      this.queue.add(() => this.processMessage(checkRunId, subtopic, message))
+      this.queue.add(() => this.processMessage(sequenceId, subtopic, message))
     })
-    await socketClient.subscribe(`account/${this.accountId}/ad-hoc-check-results/${checkRunSuiteId}/+/+`)
+    await socketClient.subscribe(`account/${this.accountId}/ad-hoc-check-results/${checkRunSuiteId}/+/+/+`)
   }
 
-  private async processMessage (checkRunId: string, subtopic: string, message: any) {
-    if (!this.timeouts.has(checkRunId)) {
+  private async processMessage (sequenceId: SequenceId, subtopic: string, message: any) {
+    if (!sequenceId) {
+      // There should always be a sequenceId, but let's have an early return to make forwards-compatibility easier.
+      return
+    }
+
+    if (!this.timeouts.has(sequenceId)) {
       // The check has already timed out. We return early to avoid reporting a duplicate result.
       return
     }
 
-    if (!this.checks.get(checkRunId)) {
-      // The check has no checkRunId associated.
+    if (!this.checks.get(sequenceId)) {
       return
     }
 
-    const { check, testResultId } = this.checks.get(checkRunId)!
+    const { check } = this.checks.get(sequenceId)!
     if (subtopic === 'run-start') {
-      this.emit(Events.CHECK_INPROGRESS, check, checkRunId)
-    } else if (subtopic === 'run-end') {
-      this.disableTimeout(checkRunId)
-      const { result } = message
+      this.emit(Events.CHECK_INPROGRESS, check, sequenceId)
+    } else if (subtopic === 'result') {
+      const { result, testResultId, resultType } = message
       await this.processCheckResult(result)
       const links = testResultId && result.hasFailures && await this.getShortLinks(testResultId)
-      this.emit(Events.CHECK_SUCCESSFUL, checkRunId, check, result, links)
-      this.emit(Events.CHECK_FINISHED, check)
+      if (resultType === 'FINAL') {
+        this.disableTimeout(sequenceId)
+        this.emit(Events.CHECK_SUCCESSFUL, sequenceId, check, result, testResultId, links)
```

Comment:
Before, we would pass the `testResultId` to the reporters at the start of the test session. 

With retries, though, we don't actually know the `testResultId` of the final test result until all of the retries are finished. Need to switch to passing it on the `CHECK_SUCCESSFUL` event.

The `testResultId` is used to generate links to the test result in the reporters.